course04/meeting02/AdamK/sortedarray/sortedarray.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+#include <algorithm>
+
+class SortedArray
+{
+public:
+	SortedArray(std::size_t maxSize);
+	void push_back(int element);
+	int operator[](std::size_t index);
+	int at(std::size_t index);
+	bool empty();
+	std::size_t size();
+	std::size_t max_size();
+	std::vector<int>::iterator begin();
+	std::vector<int>::iterator end();
```

Comment:
This isnt a valid implementation for iterators, but I will ignore this part for now, we will come back to it later